### ISSUE TITLE
npm install phantomjs-prebuilt doesn't use the global installed phantomjs

### ISSUE THREAD

**Issue by benediktarnold** [Medium/phantomjs#482]
From the [documentation](https://github.com/Medium/phantomjs#using-phantomjs-from-disk) I understand that installing phantomjs-prebuilt locally will detect phantomjs on the path and will use this instead of dowloading again. I checked this in a fresh docker container and this seems to fail.

1. Install phantomjs-prebuilt globally. Works pretty well.
2. Install phantomjs-prebuild locally. Should detect the globally installed phantomjs, but it downloads and installs locally again.

*Steps to reproduce*:
```
docker run -it node:4 bash
```
Inside the container:
```
root@e6a52800c4d1:/project# node --version
v4.3.2
root@e6a52800c4d1:/project# npm --version
2.14.12
root@e6a52800c4d1:/project# npm install -g phantomjs-prebuilt
npm info it worked if it ends with ok
...
npm info install phantomjs-prebuilt@2.1.4

> phantomjs-prebuilt@2.1.4 install /usr/local/lib/node_modules/phantomjs-prebuilt
> node install.js

Looks like an `npm install -g`; unable to check for already installed version.
Downloading https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-2.1.1-linux-x86_64.tar.bz2
Saving to /usr/local/lib/node_modules/phantomjs-prebuilt/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2
Receiving...
  [====================================----] 91%
Received 22866K total.
Extracting tar contents (via spawned process)
Removing /usr/local/lib/node_modules/phantomjs-prebuilt/lib/phantom
Copying extracted folder /usr/local/lib/node_modules/phantomjs-prebuilt/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2-extract-1457013959266/phantomjs-2.1.1-linux-x86_64 -> /usr/local/lib/node_modules/phantomjs-prebuilt/lib/phantom
Writing location.js file
Done. Phantomjs binary available at /usr/local/lib/node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs
```
Now I have a global phantomjs installation on the PATH
```
root@e6a52800c4d1:/project# echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
root@e6a52800c4d1:/project# which phantomjs
/usr/local/bin/phantomjs
```

Now the local installation
```
root@e6a52800c4d1:/project# npm install phantomjs-prebuilt
npm info it worked if it ends with ok
...
npm info install phantomjs-prebuilt@2.1.4

> phantomjs-prebuilt@2.1.4 install /project/node_modules/phantomjs-prebuilt
> node install.js

Looks like an `npm install -g`; unable to check for already installed version.
Downloading https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-2.1.1-linux-x86_64.tar.bz2
Saving to /project/node_modules/phantomjs-prebuilt/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2
Receiving...
  [====================================----] 91%
Received 22866K total.
Extracting tar contents (via spawned process)
Removing /project/node_modules/phantomjs-prebuilt/lib/phantom
Copying extracted folder /project/node_modules/phantomjs-prebuilt/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2-extract-1457014006439/phantomjs-2.1.1-linux-x86_64 -> /project/node_modules/phantomjs-prebuilt/lib/phantom
Writing location.js file
Done. Phantomjs binary available at /project/node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs
```
As you see the local install downloads again.

Is this a bug or do I misunderstand the [documentation](https://github.com/Medium/phantomjs#using-phantomjs-from-disk)?

**Comment by nicks** [comment]
ya, this is expected. it won't use an install previously installed with npm -g. The documentation says `you can install the phantomjs binary on PATH`, which is not the same thing.

**Comment by benediktarnold** [comment]
Thanks for the hint. Putting it on the path manually works.
But in my opinion the documentation is a bit unclear since npm install -g puts phantomjs on the path.

**Comment by nicks** [comment]
out of curiosity, why are you even trying to do this?

that piece of the instruction manual is explicitly for people who can't get the normal install path to work.

If you do a `npm install -g`, and then a separate `npm install`, we use different installation directories...because that's really the only logical thing to do. Otherwise, why bother running `npm install` at all? Trying to re-use the same installation directory would break npm's own consistency guarantees (for example, you could independently do a `npm uninstall -g`)

**Comment by benediktarnold** [comment]
TL;DR
I want to build a docker image with tools like phantomjs preinstalled. The image build is done with internet access. Then, on the corporate jenkins without internet connectivity (only access to a docker registry proxy, an npm proxy and a maven proxy), I want to run different project builds in containers derived from that image. These builds should use the preinstalled tools from the image.

The projects have a devDependency to phantomjs-prebuilt as a convenience for developers and an npm install would install this dependency locally.

There is nothing wrong with the way you decide if or if not to reuse the phantomjs binary - it's just that the documentation wasn't clear to me.

**Comment by swateek** [comment]
@benediktarnold too late to ask, but were you able to do it in a Dockerfile? The npm install? It's failing for me.
